docs/DeveloperGuide.md

```diff
@@ -6,17 +6,146 @@
 
 ## Design & implementation
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+Our program is structured as pictured in the following class diagram:
+![ClassDiagram.png](diagrams/ClassDiagram.png)
+
+All the inheritors of Command are features that will be described below.
+
+### Addition feature
```

Comment:
With regards to the sequence diagram, there should be a colon in front of the Class name, for example ":Duke". Because if you did not indicate, it means that you are not referring to the object. 

The "c" stated in the sequence diagram might be ambiguous too.

You might also want to the box at the bottom of the sequence diagram since it's different from the convention taught during lecture. 

The box at the bottom of the sequence diagram also can be seen causing another bug as there is no 'end' to the activation box of the Ui object at the bottom.